yaptide/celery/tasks.py

```diff
@@ -2,210 +2,197 @@
+                    "hours": int(splitted[5]),
+                    "minutes": int(splitted[7]),
+                    "seconds": int(splitted[9]),
+                }
+            }
+            stats.update(str(task_id), up_dict)
+
+        elif re.search(requested_match, line):
+            splitted = line.split(": ")
+            up_dict = {
+                "simulated_primaries": 0,
+                "requested_primaries": int(splitted[1]),
+                "task_state": SimulationModel.JobStatus.RUNNING.value
+            }
+            stats.update(str(task_id), up_dict)
+
+        elif re.search(complete_match, line):
+            splitted = line.split()
+            up_dict = {
+                "run_time": {
+                    "hours": int(splitted[2]),
+                    "minutes": int(splitted[4]),
+                    "seconds": int(splitted[6]),
+                },
+                "task_state": SimulationModel.JobStatus.COMPLETED.value
+            }
+            stats.update(str(task_id), up_dict, True)
+            return
 
 
 @celery_app.task(bind=True)
-def run_simulation(self, param_dict: dict, raw_input_dict: dict):
+def run_simulation(self, json_data: dict):
     """Simulation runner"""
     # create temporary directory
     with tempfile.TemporaryDirectory() as tmp_dir_path:
 
         # digest dictionary with project data (extracted from JSON file)
         # and generate simulation input files
-        input_files = write_input_files(param_dict, raw_input_dict, Path(tmp_dir_path))
+        input_files = write_input_files(json_data, Path(tmp_dir_path))
         # we assume here that the simulation executable is available in the PATH so pymchelper will discover it
         settings = SimulationSettings(input_path=tmp_dir_path,  # skipcq: PYL-W0612
                                       simulator_exec_path=None,
                                       cmdline_opts="")
 
-        # Pymchelper uses all available cores by default
-        ntasks = param_dict["ntasks"] if param_dict["ntasks"] > 0 else None
+        ntasks = json_data["ntasks"] if json_data["ntasks"] > 0 else None
 
         runner_obj = SHRunner(jobs=ntasks,
                               keep_workspace_after_run=True,
                               output_directory=tmp_dir_path)
+        ntasks = runner_obj.jobs
 
-        self.update_state(state="PROGRESS", meta={"path": tmp_dir_path, "sim_type": param_dict["sim_type"]})
-        try:
-            is_run_ok = runner_obj.run(settings=settings)
-            if not is_run_ok:
-                raise Exception
-        except Exception:  # skipcq: PYL-W0703
-            logfile = simulation_logfile(path=Path(tmp_dir_path, "run_1", "shieldhit_0001.log"))
-            input_files = simulation_input_files(path=tmp_dir_path)
-            return {"logfile": logfile, "input_files": input_files}
+        logs_list = [Path(tmp_dir_path) / f"run_{1+i}" / f"shieldhit_{1+i:04d}.log" for i in range(ntasks)]
```

Comment:
why not using glob ?
```suggestion
        logs_list = [Path(tmp_dir_path).glob("run_*/shieldhit_*log")]
```

Later check if you got exactly the same number of logs as expected

Reply:
It won't find anything because files are not created yet -> creation is handled by Pymchelper in 164 line

Reply:
You are correct !